test/client/spec/victory-util/add-events.spec.js

```diff
@@ -0,0 +1,95 @@
+/* eslint max-params: 0 */
+
+import React from "react";
+import { forEach, get } from "lodash";
+import { mount } from "enzyme";
+import { addEvents } from "src/index";
+import { MockChart, MockLabel, MockDataComponent } from "../mock-components";
+
+describe("victory-util/add-events", () => {
+  const EventedMockChart = addEvents(MockChart);
+
+  const expectEventsTriggered = (scopeFn, testFn, expectations, wrapper) => {
```

Comment:
Basically, this function is "Take a wrapper, get an array of components with `scopeFn`, map those components with `testFn`, and expect that each test result equals the corresponding expectation". I think I could rephrase this more as a pipeline if it would be more clear that way. Basically it just allows a comprehensive list of conditions to be tested on a set of components. 

I've seen the examples of using ` eventHandlers: { onClick: spy }`, the pattern being to trigger an event and then check that `spy` was called with the signature `(evt, childProps, eventKey, this)`, with `childProps` and `eventKey` corresponding to the expected target of the event. I think this could be a bit fragile though, since it doesn't test all the way through - at the end of the test, we only know that the handler function we supplied (normally to return a config object for the event) was invoked in some relation to the expected target. We need to read the internals to know what that relation is though, and we still don't know whether anything changed in the app. 

Basically, `sinon.spy` tests are satisfied here: https://github.com/FormidableLabs/victory-core/blob/master/src/victory-util/events.js#L163

We could introduce a bug in the state change or even nuke this line of code and break the entire events system, and our current tests using `sinon.spy` would all pass since they're not able to see that far. https://github.com/FormidableLabs/victory-core/blob/master/src/victory-util/events.js#L166

That's why I opted to check the final result instead - it decouples our tests from internal details and gets full testing of a feature. 

Thoughts?